lib/features/profile/data/datasources/profile_service.dart

```diff
@@ -0,0 +1,32 @@
+import 'dart:io';
+import 'package:dio/dio.dart';
+import 'package:final_project/core/data/models/user_model.dart';
+import 'package:final_project/features/profile/data/models/image_model.dart';
+import 'package:injectable/injectable.dart';
+import 'package:retrofit/retrofit.dart';
+part 'profile_service.g.dart';
+
+@lazySingleton
```

Comment:
There is a layer missing,

`DataSource` uses `Service`

`Repository` uses `DataSource`

`UseCase` uses `Repository`

`Cubit` uses `UseCase`